fairscale/nn/pipe/async_pipe.py

```diff
@@ -38,10 +44,171 @@ def __len__(self) -> int:
         return len(self.modules)
 
 
-class AsyncPipe(MultiProcessPipe):
-    def __init__(self, *args: Any, **kwargs: Any):
-        super().__init__(*args, **kwargs)
-        self.pipelined_backward = False
+def verify_module(module: Union[nn.Sequential, List[LazyModule]]) -> None:
+    if len(set(map(id, module))) != len(module):
+        raise ValueError("module with duplicate children is not supported")
+
+
+def check_balance(module: Union[nn.Sequential, List[LazyModule]], balance: List[int]) -> None:
+    if len(module) != sum(balance):
+        raise ValueError(
+            f"module and sum of balance have different length (module: {len(module)}, sum of balance: {sum(balance)})"
+        )
+
+    if any(x <= 0 for x in balance):
+        raise ValueError(f"all balance numbers must be positive integer (balance: {balance})")
+
+
+MOVING_DENIED = TypeError("denied to move parameters and buffers, because Pipe should manage device placement")
+
+
+class AsyncPipe(Module):
+    """Wraps an arbitrary :class:`nn.Sequential <torch.nn.Sequential>` module
+    to train on Pipe_. If the module requires lots of memory, Pipe will be
+    very efficient.
+    ::
+
+        model = nn.Sequential(a, b, c, d)
+        model = Pipe(model, balance=[1, 1, 1, 1], chunks=8)
+        output = model(input)
+
+    .. _Pipe: https://arxiv.org/abs/1811.06965
```

Comment:
Would be good to have this docstring mention what is different between MultiprocessPipe and AsyncPipe? I don't know if this doc string is relevant anymore.